ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs

```diff
@@ -909,6 +926,20 @@ stdLowLevelRunNodeArgsIO RunNodeArgs{ rnProtocolInfo
       , llrnMaxCaughtUpAge = secondsToNominalDiffTime $ 20 * 60   -- 20 min
       , llrnMaxClockSkew =
           InFuture.defaultClockSkew
+      , llrnMaxRollbackAge = case gcCSJConfig rnGenesisConfig of
+          -- If CSJ is disabled, we are either syncing from trusted peers (Praos
+          -- mode), or the LoP already handles historical rollbacks.
+          CSJDisabled  -> Nothing
+          -- If CSJ is enabled, we prevent historical rollbacks: When peers roll
+          -- back before their last acknowledged jump, we disengage them
+          -- (causing us to run full ChainSync with them). This is fine when we
+          -- are almost caught-up, but it is wasteful while syncing historical
+          -- parts of the chain.
+          --
+          -- Here, "historical" is defined to be the length of a Shelley
```

Comment:
Perhaps add a comment explaining that this magic number is specific to Cardano mainnet.

Seems like the duration itself should be an `llrn` parameter, at least for the sake of nodes that aren't running Cardano mainnet (eg testnets). If so, that parameter should explain that the user should choose a duration that is slightly larger than the largest Chain Growth property's window of any era of their chain.